src/RecursiveStagesInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace SilverStripe\Versioned;
+
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Interface RecursiveStagesInterface
+ *
+ * Interface for @see RecursiveStagesService to provide the capability to for "smart" durty model state
```

Comment:
```suggestion
 * Interface for @see RecursiveStagesService to provide the capability to for "smart" dirty model state
```
What is "smart" dirty model state?

Reply:
It's a dirty model state that traverses down to owned models.